staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1/api.proto

```diff
@@ -185,6 +194,8 @@ message ContainerAllocateResponse {
 	repeated DeviceSpec devices = 3;
 	// Container annotations to pass to the container runtime
 	map<string, string> annotations = 4;
+	// CDI devices for the container.
+	repeated CDIDevice cdi_devices = 5;
 }
```

Comment:
This is an API change as far as I understand. Did you test if current/old plugins still work with it?

Reply:
I need to still dig through the tests for the device plugin. I will update the PR with details once I have better insight into this.

Reply:
It would be good to cover this functionality with unit and e2e tests.

Reply:
Yes, I had a look at the existing unit tests today and they would require some work before adding new ones would be simple. I am also looking into the e2e tests and will definitely add them there.

Reply:
I also agree (but it seems there's general consensus already) adding unit tests would be nice. Probably the best way bar only way here is extending `sample-device-plugin`. We need to document we need to keep variants with/without `CDIDevices` reporting until the feature is GA. Will require some coordination but is doable.
I think we should start the e2e work and call it a prerequisite for the beta graduation but I don't think it should block alpha. I recon setting up the bits and pieces to enable e2e testing will take some time.

Reply:
It is not an API change in the sense that we need a version bump. Additional fields added to a protobuf don't break the API for components that are built against an earlier revision of the proto spec without the additional field.